website/js/FactSheet.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+
+var React = require('react');
+var PureRenderMixin = require('./PureRenderMixin'); // deep-equals version of PRM
+var {Grid, Col, Row, Alert} = require('react-bootstrap');
+var backend = require('backend');
+var {Link} = require('react-router');
+
+var FactSheet = React.createClass({
+    mixins: [PureRenderMixin],
+    getInitialState: function() {
+        return {};
+    },
+    componentWillMount: function() {
+        backend.variantCounts().subscribe(
+            resp => this.setState(this.setState(resp)),
```

Comment:
never seen this before (setState inside a setState call), is this intentional and necessary?